projects/packages/forms/src/class-rest-controller.php

```diff
@@ -0,0 +1,247 @@
+				sprintf( esc_html__( '%s must be a non-negative integer.', 'jetpack-forms' ), $param )
+			);
+		}
+		return true;
+	}
+
+	/**
+	 * Validates that the parameter is a string.
+	 *
+	 * @since 4.3.0
+	 *
+	 * @param string          $value Value to check.
+	 * @param WP_REST_Request $request The request sent to the WP REST API.
+	 * @param string          $param Name of the parameter passed to endpoint holding $value.
+	 *
+	 * @return bool|WP_Error
+	 */
+	public static function validate_string( $value, $request, $param ) {
+		if ( ! is_string( $value ) ) {
+			return new WP_Error(
+				'invalid_param',
+				sprintf(
+					/* Translators: Placeholder is a parameter name. */
+					esc_html__( '%s must be a string.', 'jetpack-forms' ),
+					$param
+				)
+			);
+		}
+		return true;
+	}
+
+	/**
+	 * Returns Jetpack Forms responses.
+	 *
+	 * @param WP_REST_Request $request The request sent to the WP REST API.
+	 *
+	 * @return WP_REST_Response A response object containing Jetpack Forms responses.
+	 */
+	public function get_jetpack_form_responses( $request ) {
+		$args = array(
+			'post_type'   => 'feedback',
+			'post_status' => array( 'publish', 'draft' ),
+		);
+
+		if ( isset( $request['form_id'] ) ) {
+			$args['post_parent'] = $request['form_id'];
+		}
+
+		if ( isset( $request['limit'] ) ) {
+			$args['posts_per_page'] = $request['limit'];
+		}
+
+		if ( isset( $request['offset'] ) ) {
+			$args['offset'] = $request['offset'];
+		}
+
+		if ( isset( $request['search'] ) ) {
+			$args['s'] = $request['search'];
+		}
+
+		$query = new \WP_Query( $args );
+
+		$responses = array_map(
+			function ( $response ) {
+				$data = \Automattic\Jetpack\Forms\ContactForm\Contact_Form_Plugin::parse_fields_from_content( $response->ID );
+
+				return array(
+					'id'                      => $response->ID,
+					'uid'                     => $data['all_fields']['feedback_id'],
+					'date'                    => get_the_date( 'c', $data ),
+					'author_name'             => $data['_feedback_author'],
+					'author_email'            => $data['_feedback_author_email'],
+					'author_url'              => $data['_feedback_author_url'],
+					'author_avatar'           => empty( $data['_feedback_author_email'] ) ? '' : get_avatar_url( $data['_feedback_author_email'] ),
+					'email_marketing_consent' => $data['all_fields']['email_marketing_consent'],
+					'ip'                      => $data['_feedback_ip'],
+					'entry_title'             => $data['all_fields']['entry_title'],
+					'entry_permalink'         => $data['all_fields']['entry_permalink'],
+					'subject'                 => $data['_feedback_subject'],
+					'fields'                  => array_diff_key(
+						$data['all_fields'],
+						array(
+							'email_marketing_consent' => '',
+							'entry_title'             => '',
+							'entry_permalink'         => '',
+							'feedback_id'             => '',
+						)
+					),
+				);
+			},
+			$query->posts
+		);
+
+		return rest_ensure_response(
```

Comment:
This is a good starting point, although it's worth noting that the designs have these things:

![Screenshot 2023-02-20 at 18 38 45](https://user-images.githubusercontent.com/8056203/220171455-532646be-f47d-422c-9b8c-1e4f54efad61.png)

So we'll probably need to add totals for all the categories, not just the current query. Just wanted to note that - we can add a separate task for it.

Reply:
Yes, a lot of things are still missing. Again, amidst all the things I was trying, I copied what was already functional from the old PR. Let's use this one to settle on the building blocks and address this on next iterations?